app/controllers/microposts_controller.rb

```diff
@@ -0,0 +1,92 @@
+class MicropostsController < ApplicationController
+  before_action :set_micropost, only: %i(show edit update destroy)
+
+  # GET /microposts or /microposts.json
+  def index
+    @microposts = Micropost.all
+  end
+
+  # GET /microposts/1 or /microposts/1.json
+  def show; end
+
+  # GET /microposts/new
+  def new
+    @micropost = Micropost.new
+  end
+
+  # GET /microposts/1/edit
+  def edit; end
+
+  # POST /microposts or /microposts.json
+  def create
+    @micropost = Micropost.new(micropost_params)
+
+    respond_to do |format|
+      if @micropost.save
+        format.html do
+          redirect_to micropost_url(@micropost),
+                      notice: t("microposts.created")
+        end
+        format.json{render :show, status: :created, location: @micropost}
+      else
+        format.html{render :new, status: :unprocessable_entity}
+        format.json do
+          render json: @micropost.errors,
+                 status: :unprocessable_entity
+        end
+      end
+    end
+  end
+
+  # PATCH/PUT /microposts/1 or /microposts/1.json
+  def update
+    respond_to do |format|
+      if @micropost.update(micropost_params)
+        format.html do
+          redirect_to micropost_url(@micropost),
+                      notice: t("microposts.updated")
+        end
+        format.json{render :show, status: :ok, location: @micropost}
```

Comment:
```suggestion
        format.json {render :show, status: :ok, location: @micropost}
```
Thêm 1 space trước block {} e nha
Fix tương tự những chỗ khác